source/main/modules/window-manager/index.ts

```diff
@@ -207,18 +223,72 @@ export default class WindowManager {
   /**
    * Shuts down the window manager and performs final operations
    */
-  shutdown (): void {
+  shutdown(): void {
     this._persistWindowPositions()
   }
 
   /**
    * Listens to events on the main window
    */
-  private _hookMainWindow (): void {
+  private _hookMainWindow(): void {
     if (this._mainWindow === null) {
       return
     }
 
+    this._mainWindow.on('show', async () => {
+      if (process.platform == 'darwin') {
+        if (tray == null) {
+          let basepath = app.getAppPath()
+          let image = await nativeImage.createThumbnailFromPath(path.join(__dirname, 'assets/icons/128x128.png'), { width: 16, height: 16 })
+          tray = new Tray(image)
+          const contextMenu = Menu.buildFromTemplate([
+            {
+              label: 'Show Zettlr',
+              click: () => {
+                this.showAnyWindow()
+              },
+              type: 'normal'
+            },
+            { label: '', type: 'separator' },
+            {
+              label: 'Quit',
+              click: () => {
+                app.quit()
+              }, type: 'normal'
+            }
+          ])
+          tray.setToolTip('This is zettlr tray. \n click show Zettlr button to display app \n click quit to quit app')
+          tray.setContextMenu(contextMenu)
+        }
+      } else {
+        if (tray == null) {
+          tray = new Tray(path.join(__dirname, 'assets/icons/128x128.png'))
+          const contextMenu = Menu.buildFromTemplate([
+            {
+              label: 'Show Zettlr',
+              click: () => {
+                // Add show Zettlr window event to Windows #6
+                this.showMainWindow()
+              },
+              type: 'normal'
+            },
+            { label: '', type: 'separator' },
+            {
+              label: 'Quit',
+              click: () => {
+                // Add quit event to tray #8
+                // On Windows, left or right click the tray icon ➔ Quit will quit Zettlr. Same function as File ➔ Quit.
+                app.quit()
```

Comment:
The tray isn't working for me so I can't test at the moment, but does `app.quit()` follow this part of the [tray design](https://github.com/UofA-SPI21-team7/zettlr-assignment/wiki/Tray-design)?:

> If there are unsaved changes, show and bring-to-the-front the Zettlr window and display the 'There are unsaved changes' modal dialog.

 I tried to follow the  Zettlr code and if I follow correctly this will quit immediately?